File: spec/features/score_spec.rb
feature 'when the game begins', js: true do

    before(:each) { visit '/' }
    
    scenario 'it should display opening content' do
        expect(page).to have_content "Frame: 1"
        expect(page).to have_content "Roll: 1"
        expect(page).to have_select("pins", options: ['Pins knocked down', '0','1','2','3','4','5','6','7','8','9','10'])
    end

    scenario 'on selecting pins hit the total score should update' do
        select '4', from: "pins"
        expect(page).to have_content "Total Score: 4"
    end

    scenario 'on hitting a stike, score card should show x' do
        select '10', from: "pins"
        expect(page).to have_content "X"
    end

    scenario 'no X if 10 on roll 2' do
        select '0', from: "pins"
        select '10', from: "pins"
        expect(page).not_to have_content "X"
    end

    scenario 'drop down menu should update with pins left standing after a hit' do
        select '3', from: "pins"
        expect(page).to have_select("pins", options: ['Pins knocked down', '0','1','2','3','4','5','6','7'])
    end

end

File: spec/features/scoreSpec.js
describe("Score tests: ", function() {
  var game;

  beforeEach(function() {
    game = new Game();
  });

  describe("#score:", function() {
    it('should have a score of 10 after a strike', function() {
      game.go(5);
      expect(game.score()).toEqual(5);
    });

    it("a 'random' generation of X19828160XXX1933", function() {
      [10, 1, 9, 8, 2, 8, 1, 6, 0, 10, 10, 10, 1, 9, 3, 3].forEach(function(i) {
        game.go(i);
      });
      expect(game.score()).toEqual(161);
    });

    it("a 'random' generation of X19828160XXX19XXX", function() {
      [10, 1, 9, 8, 2, 8, 1, 6, 0, 10, 10, 10, 1, 9, 10, 10, 10].forEach(function(i) {
        game.go(i);
      });
      expect(game.score()).toEqual(222);
    });

    it("a 'random' generation of XXXXXXXXXXXX", function() {
      [10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10].forEach(function(i, index) {
        console.log('BEFORE go, i: ' + i + ", score = " + game.score() + ", index: " + index);
        game.go(i);
      });
      expect(game.score()).toEqual(300);
    });
  });
});
